### ISSUE TITLE
Identity Ground Truth Correspondence Matrix

### ISSUE THREAD

**Issue by sean-xr** [riccardomarin/Diff-FMAPs-PyTorch#3]
Dear Author,

Thanks for the great work! The code is well organized and the documentation is really well written as well.

However, I have one difficulty in understanding your code, in train_basis.py, the loss is written as: 
        eucl_loss = torch.sum(torch.square(torch.matmul(s_max_matrix, torch.transpose(pc_B,1,2)) - torch.transpose(pc_B,1,2)))

While in the paper, the same loss function is formulated as: 
![Screenshot 2023-12-04 at 15 56 29](https://github.com/riccardomarin/Diff-FMAPs-PyTorch/assets/91930856/e18bf9cf-ed88-4bff-99aa-f71e5b35bdf4)

So why are we assuming the ground-truth correspondence in the code to be identity?

Thanks for the answer.

**Comment by riccardomarin** [comment]
Hi @sean-xr ,
Thanks for your nice words and for reaching out!

You are correct: in the code, the GT correspondence is assumed to be the identity at training time. The reason is that we trained the network on SMPL models, all with the same vertex order, and so we can use this simplified version.

Having a code dealing with arbitrary GT correspondence would be more general (e.g., to train on remeshed versions of FAUST and SCAPE datasets), but for our experiments, it was not needed.

Let me know if this clarifies your doubts :)

Best!

**Comment by sean-xr** [comment]
Thanks for the prompt answer. It clears my doubt, wish you all the best
